### DISCUSSION TITLE
Tracking mouse activity on view

### DISCUSSION THREAD

**Discussion by dasien** [tui-cs/Terminal.Gui#3414] [Q&A] [answered]
Hello,

I am using v 1.16 and i have a question about tracking a mouse event.  I have a timer that runs on the `MainLoop` that will lock the application if there is no mouse or keyboard activity after a certain amount of time.  In my main view, i have overridden the `OnKeyDown` and `OnMouseEvent` methods, flipping a "keep alive" bit so that when the timer fires, it knows that there has been activity and not to lock the application.

The main view has just a few controls on it (treeview and listview).

I am able to catch the `KeyDown` events in my override, but i do not see any `MouseEvents` being thrown to that  main view.  

How can i also capture mouse movements, clicks, etc.?  Do i need to implement those overrides on the sub-views as well?

**Comment by BDisp** [comment, marked as answer]
You can subscribe to the `Application.RootMouseEvent` event to capture any mouse event.

**Comment by dasien** [comment]
Interestingly, i tried to just create event handlers rather than overriding the virtual methods above. I put event handlers in for `KeyPress` and `MouseClick`.  I tried to attach these to both the `TopLevel` and then my main `Window`.  Neither of these event handlers are ever fired for any keyboard press or mouse clicks on the `SubViews` of the `Window`.

The only thing that i can get consistently to fire is the override of `KeyDown` on the `TopLevel`.
